File: app/src/main/java/com/androidwind/androidquick/sample/MyApplication.java
package com.androidwind.androidquick.sample;

import androidx.multidex.MultiDexApplication;

/**
 * @author ddnosh
 * @website http://blog.csdn.net/ddnosh
 */
public class MyApplication extends MultiDexApplication {

    private static MyApplication INSTANCE;

    @Override
    public void onCreate() {
        super.onCreate();
        //get application
        if (INSTANCE == null) {
            INSTANCE = this;
        }
    }

    public static synchronized MyApplication getInstance() {
        return INSTANCE;
    }
}

File: app/src/main/java/com/androidwind/androidquick/sample/MyApplication.kt
package com.androidwind.androidquick.sample

import android.content.Context
import androidx.multidex.MultiDexApplication
import com.androidwind.androidquick.sample.util.TimeUtils
import com.androidwind.androidquick.util.ToastUtil


/**
 * @author ddnosh
 * @website http://blog.csdn.net/ddnosh
 */
class MyApplication : MultiDexApplication() {
    override fun attachBaseContext(base: Context?) {
        super.attachBaseContext(base)
//        if (isMainProcess()) { //判断主线程
        TimeUtils.beginTimeCalculate(TimeUtils.COLD_START);
//        }
    }

    override fun onCreate() {
        super.onCreate()
        //get application
        instance ?: run {
            instance = this
        }
        ToastUtil.register(this)
    }

    companion object {
        @get:Synchronized
        var instance: MyApplication? = null
    }
}